File: frontend/src/types/tenant.ts
export interface Tenant {
  id: number;
  property_id: number;
  first_name: string;
  last_name: string;
  email: string;
  phone: string;
  move_in_date: string | null;
  move_out_date: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface TenantCreateRequest {
  property_id: number;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  move_in_date?: string;
}

export interface TenantUpdateRequest {
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  move_in_date?: string;
  move_out_date?: string;
  is_active?: boolean;
}

File: backend/internal/models/tenant.go
package models

import (
	"database/sql"
	"time"
)

type Tenant struct {
	ID          int          `json:"id"`
	PropertyID  int          `json:"property_id"`
	FirstName   string       `json:"first_name"`
	LastName    string       `json:"last_name"`
	Email       *string      `json:"email,omitempty"`
	Phone       *string      `json:"phone,omitempty"`
	MoveInDate  sql.NullTime `json:"move_in_date"`
	MoveOutDate sql.NullTime `json:"move_out_date"`
	IsActive    bool         `json:"is_active"`
	CreatedAt   time.Time    `json:"created_at"`
	UpdatedAt   time.Time    `json:"updated_at"`
}

type TenantCreateRequest struct {
	PropertyID int    `json:"property_id" binding:"required"`
	FirstName  string `json:"first_name" binding:"required"`
	LastName   string `json:"last_name" binding:"required"`
	Email      string `json:"email"`
	Phone      string `json:"phone"`
	MoveInDate string `json:"move_in_date"` // Format: "2006-01-02"
}

type TenantUpdateRequest struct {
	FirstName   string `json:"first_name"`
	LastName    string `json:"last_name"`
	Email       string `json:"email"`
	Phone       string `json:"phone"`
	MoveInDate  string `json:"move_in_date"`
	MoveOutDate string `json:"move_out_date"`
	IsActive    *bool  `json:"is_active"`
}
